src/main/java/pl/javastart/task/services/MixPhoneContract.java

```diff
@@ -0,0 +1,104 @@
+package pl.javastart.task.services;
+
+public class MixPhoneContract extends CardPhoneContract {
+    private int remainingSms;
+    private int remainingMms;
+    private double remainingMinutes;
+
+    public MixPhoneContract(double balance, double smsFee, double mmsFee, double minuteCallFee,
+                            int smsAvailable, int mmsAvailable, double minutesAvailable) {
+        super(balance, smsFee, mmsFee, minuteCallFee);
+        this.remainingSms = smsAvailable;
+        this.remainingMms = mmsAvailable;
+        this.remainingMinutes = minutesAvailable;
+    }
+
+    public int getRemainingSms() {
+        return remainingSms;
+    }
+
+    protected void setRemainingSms(int remainingSms) {
+        this.remainingSms = remainingSms;
+    }
+
+    public int getRemainingMms() {
+        return remainingMms;
+    }
+
+    protected void setRemainingMms(int remainingMms) {
+        this.remainingMms = remainingMms;
+    }
+
+    public double getRemainingMinutes() {
+        return remainingMinutes;
+    }
+
+    protected void setRemainingMinutes(double remainingMinutes) {
+        this.remainingMinutes = remainingMinutes;
+    }
+
+    @Override
+    protected int availableCallSeconds(int seconds) {
+        double remainingMinutesAsSeconds = (int) (remainingMinutes * 60);
+        double allowanceSeconds = 0;
+        if (remainingMinutesAsSeconds >= seconds) { // pobiera wszystko z darmowych minut
+            callInSeconds = callInSeconds + seconds;
+            allowanceSeconds = seconds;
+            remainingMinutes -= allowanceSeconds / 60;
+            return (int) allowanceSeconds;
+        } else if (remainingMinutesAsSeconds > 0 && remainingMinutesAsSeconds < seconds) { // wybiera reszte z darmowych minut
+            remainingMinutes = 0;
+            callInSeconds = (int) (callInSeconds + remainingMinutesAsSeconds);
+            allowanceSeconds += remainingMinutesAsSeconds;
+        }
+        double feeForSecond = minuteCallFee / 60;
+        int secondsHasToBePayedFromCredit = (int) (seconds - remainingMinutesAsSeconds);
+        double feeForSecondsFromCredit = secondsHasToBePayedFromCredit * feeForSecond;
+
+        if (creditBalance >= feeForSecondsFromCredit) { // wystarczająca ilość creditu
+            creditBalance = creditBalance - feeForSecondsFromCredit;
+            callInSeconds = callInSeconds + secondsHasToBePayedFromCredit;
+            allowanceSeconds += secondsHasToBePayedFromCredit;
+            return (int) allowanceSeconds;
+        }
+
+        int creditRating = (int) (creditBalance / feeForSecond);
+
+        if (creditBalance > 0) { // dostępna tylko część creditu
+            creditBalance -= creditRating * feeForSecond;
+            callInSeconds = (callInSeconds + creditRating);
+            allowanceSeconds += creditRating;
+        }
+        return (int) allowanceSeconds;
```

Comment:
Biorąc pod uwagę fakt, że dziedziczysz tutaj po klasie `CardPhoneContract`, to powinieneś to w jakiś sposób wykorzystać, aby nie zapisywać tej samej logiki dwa razy.
Algorytm tej metody to:
1. Wykorzystaj pakiet minut
2. Jeżeli pakiet minut jest niewystarczający, to wykorzystaj stan konta (metoda availableCallSeconds z klasy nadrzędnej)
3. Jeżeli stan konta się wyczerpał, to zakończ rozmowę